middleware/http/server.go

```diff
@@ -101,6 +110,17 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// try to extract B3 Headers from upstream
 	sc := h.tracer.Extract(b3.ExtractHTTP(r))
 
+	// store whitelisted headers to be propagated in spanContext
+	if h.baggage != nil {
+		sc.Baggage = h.baggage.Init()
+		sc.Baggage.IterateWhiteList(func(key string) {
+			vals := r.Header.Values(key)
+			if len(vals) > 0 {
+				sc.Baggage.AddHeader(key, vals...)
```

Comment:
too late to change the api I guess, but since at least in HTTP multi-value headers are defined strictly based on the header name and also have concatenation rules.. it might be easier and less allocation on the whole to have a single value api. ex you'd not worry about the check val length part, and those serializing would know the specific key has encoding rules as they are the one using it.. so force them to do that instead of the api.

not interested in debating this, just injecting it into your head as it makes the api easier and no need for utilities after!

Reply:
I took the approach of baggage being purely propagation driven, as in... new headers are not injected into Baggage but rather they originate from being picked up in server middleware. The whole baggage thingy is mostly used in transport middleware (where we already provide http and grpc) and not so much meant for direct api access. Yeah people will abuse... they'll always find a way. At least this way it fits with both Go's HTTP header and gRPC metadata multi value items. (1:1 transport)